src/FormElement/ScheduleElement/Recurrence.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace ipl\Web\FormElement\ScheduleElement;
+
+use DateTime;
+use ipl\Html\Attributes;
+use ipl\Html\FormElement\BaseFormElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\HtmlString;
+use ipl\I18n\Translation;
+use ipl\Scheduler\Contract\Frequency;
+use ipl\Scheduler\RRule;
+
+class Recurrence extends BaseFormElement
+{
+    use Translation;
+
+    protected $tag = 'div';
+
+    protected $defaultAttributes = ['class' => 'schedule-occurrence'];
+
+    /** @var callable A callable that generates a frequency instance */
+    protected $frequencyCallback;
+
+    /** @var callable A validation callback for the schedule element */
+    protected $isValidCallback;
+
+    /**
+     * Set a validation callback that will be called when assembling this element
+     *
+     * @param callable $callback
+     *
+     * @return $this
+     */
+    public function setValid(callable $callback): self
+    {
+        $this->isValidCallback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Set a callback that generates an {@see Frequency} instance
+     *
+     * @param callable $callback
+     *
+     * @return $this
+     */
+    public function setFrequency(callable $callback): self
+    {
+        $this->frequencyCallback = $callback;
+
+        return $this;
+    }
+
+    protected function assemble()
+    {
+        $isValid = ($this->isValidCallback)();
+        if (! $isValid) {
+            return;
+        }
+
+        /** @var RRule $frequency */
+        $frequency = ($this->frequencyCallback)();
+        $recurrences = $frequency->getNextRecurrences(new DateTime(), 3);
+        if (! $recurrences->valid()) {
+            // Such a situation can be caused by setting an invalid end time
+            $this->addHtml(HtmlString::create($this->translate('Recurrences cannot be generated')));
```

Comment:
That's not a HTML string. It's text. Use `ipl\Html\Text`. Otherwise, a malicious translation is not escaped.